packages/git/src/browser/git-repository-provider.ts

```diff
@@ -93,6 +94,9 @@ export class GitRepositoryProvider {
      * Sets or un-sets the repository.
      */
     set selectedRepository(repository: Repository | undefined) {
+        if (!repository) {
+            this.scmService.selectedRepository = repository;
+        }
         this._selectedRepository = repository;
```

Comment:
- What is the difference between `selectedRepository` in `GitRepositoryProvider` and `ScmService`?
- Should not they point to the same repository? Can `GitRepositoryProvider` to `ScmService` for `selectedRepository` and `onDidChangeRepository` event?